### ISSUE TITLE
MueLu: PermutationFactory is untested

### ISSUE THREAD

**Issue by GrahamBenHarper** [trilinos/Trilinos#11836]
As mentioned in a thread on #11821 and https://github.com/trilinos/Trilinos/issues/11821#issuecomment-1528064787, `PermutationFactory` is currently untested because the logic explicitly requires Tpetra=OFF, but we've required Tpetra since #9532. The test is written using Epetra, and a trivial change in CMake logic makes it test when Epetra testing runs. Our current available options are:
1. Test it
2. Delete it

I tend to lean toward option 1., and I'd like it to be tested on the Tpetra stack as well, but it's unclear how much work would be at the moment. I did some tinkering with that in the referenced comment but it still didn't work. I'll open a PR with the corresponding changes today at some point.

**Comment by jhux2** [comment]
Does anyone use this feature?

**Comment by GrahamBenHarper** [comment]
@jhux2 mentioned in https://github.com/trilinos/Trilinos/pull/11830#issuecomment-1527140085, it seems like nobody we know uses it. It would reduce technical debt to delete it, but I don't have a great understanding of what it does, so I feel like I'm only qualified to add tests and not to delete it.

**Comment by mayrmt** [comment]
@maxfirmbach Yes, that is the use case by @tawiesn. We currently don't have any applications for this. I want to talk to @poppmuc to clarify some remaining open questions, before making a final statement about the necessity of this factory.

**Comment by GrahamBenHarper** [comment]
@mayrmt is this still in use?

**Comment by mayrmt** [comment]
@GrahamBenHarper, from our side, these can go away.
